cmd/analyse.go

```diff
@@ -49,23 +50,26 @@ func destructor(cmd *cobra.Command, args []string) {
 				// If file doesn't exists, continue
 				continue
 			}
+			return err
 		}
 		e := os.Remove(file)
 		if e != nil {
-			log.Fatal().Msgf("Error clearing files %s", file)
+			return fmt.Errorf("error clearing files %s", file)
 		}
 	}
+	return nil
 }
 
 //runAnalyse is controller func for analyses cmd.
-func runAnalyse(cmd *cobra.Command, args []string) {
+func runAnalyse(cmd *cobra.Command, args []string) error {
+	telemetry.SetFlag(cmd.Context(), "json", jsonOut)
+	telemetry.SetFlag(cmd.Context(), "verbose", verboseOut)
 	if !viper.IsSet("crda_key") {
-		fmt.Fprintln(os.Stdout,
-			color.RedString("\u2718 "),
-			"Please run `crda auth` command first.",
-		)
-		os.Exit(1)
+		telemetry.SetExitCode(cmd.Context(), 1)
```

Comment:
I see that code is same across multiple errors, do we send the textual notation of the error from somewhere along with the code?

Reply:
Also what would be the value add of having error code if we are already sending the textual notation of the error?

Reply:
Error codes can have separate Graphs, which can tell us How many Exit-codes 1,2,0 our cli is throwing

Reply:
even textual notations can be treated as enumerations.

Reply:
Not sure, how can we deside on `exit-code=2` in that case. Additionally I think Parag also asked for this.

Reply:
I dont think this provides value.

Reply:
After discussion, we concluded that `exit-code` does provide value and hence we are keeping it.

Reply:
@deepak1725 after this change, when we try to get the exit value of a process, it will return me only 0,1,2 or will it also return me the string along with it?

Reply:
Ya, ya, there is no change in Behaviour

We were bit doubtful if we should record `exit-code` or not, and hence we will concluded this to keep it